crates/jupyter-protocol/src/lib.rs

```diff
@@ -0,0 +1,14 @@
+pub use jupyter_serde::*;
+pub mod messaging;
+pub use messaging::*;
+
+mod time;
+
+use async_trait::async_trait;
+use futures::{Sink, Stream};
+
+#[async_trait]
+pub trait JupyterConnection:
+    Sink<JupyterMessage> + Stream<Item = Result<JupyterMessage, anyhow::Error>>
+{
```

Comment:
I _almost_ added `connect` here then realized I could just require the `Sink` and `Stream` until I know and need more of an interface.